snapshot/store/groupcache_store.go

```diff
@@ -155,10 +159,8 @@ func (s *groupcacheStore) Write(name string, data io.Reader, ttl *TTLValue) erro
 	c := make([]byte, len(b))
 	copy(c, b)
 
-	d := GetDurationTTL(ttl)
-
 	log.Infof("Write() bundle %s: length: %d ttl: %s", name, len(b), ttl)
-	if err := s.cache.Put(nil, name, c, d); err != nil {
+	if err := s.cache.Put(nil, name, c, &ttl.TTL); err != nil {
```

Comment:
May need a defense against nil ttl parameter here

Reply:
Done.

Reply:
It's actually ok to store a nil ttl, I meant just check here before dereferencing

Reply:
ack